### PR TITLE
[Enhancement] Report error state tablet to FE and show error state tablet in statistic

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

To avoid some serious bugs, we add many check for primary key tablet and if check failed, primary key tablet state will be set error which will refuse all operation. However, the error state is not easily observable and this pr report the error state tablet to FE and we can use `show proc '/statistic/;`  to check if there are some tablet in error state.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java [hunk 1]
```diff
@@ -355,6 +358,14 @@ private boolean needSync(Replica replicaInFe, TTabletInfo backendTabletInfo) {
             return true;
         }
 
+        // error state need sync or not
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java:361]: remove the comment?

Comment (sevev) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java]: > remove the comment?

done.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 32 (12.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|8|00.00%|[175, 176, 177, 178, 179, 180, 182, 183]|
|:large_blue_circle:|com/starrocks/common/proc/ReplicasProcNode.java|0|2|00.00%|[58, 114]|
|:large_blue_circle:|com/starrocks/common/proc/IncompleteTabletsProcNode.java|0|5|00.00%|[32, 44, 48, 62, 66]|
|:large_blue_circle:|com/starrocks/catalog/Replica.java|1|6|16.67%|[319, 323, 324, 326, 327]|
|:large_blue_circle:|com/starrocks/common/proc/StatisticProcDir.java|2|9|22.22%|[116, 162, 163, 196, 226, 261, 262]|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|1|2|50.00%|[168]|

**Comment by sevev** [comment]
run starrocks_be_unittest

**Comment by sevev** [comment]
run all

File: fe/fe-core/src/main/java/com/starrocks/catalog/LocalTablet.java [hunk 2]
```diff
@@ -171,6 +171,18 @@ public void addReplica(Replica replica) {
         addReplica(replica, true);
     }
 
+    public int getErrorStateReplicaNum() {
```
[the remark this answers is not in the document]
Comment (sevev) [reply fe/fe-core/src/main/java/com/starrocks/catalog/LocalTablet.java:174]: > The `show proc '/statistic/dbId'` need to output the tablet list.

This command will show the error tablet list but not show the error replicate id, we can use show tablet $tablet_id to get which replicate is in error state.

**Review by chaoyli — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**